integration/setup/bastion_instance.go

```diff
@@ -0,0 +1,559 @@
+// +build k8srequired
+
+package setup
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/giantswarm/microerror"
+)
+
+func ensureBastionHostCreated(ctx context.Context, clusterID string, config Config) error {
+	var err error
+
+	var subnetID string
+	var vpcID string
+	{
+		config.Logger.LogCtx(ctx, "level", "debug", "message", "finding public subnet and vpc")
+
+		i := &ec2.DescribeSubnetsInput{
+			Filters: []*ec2.Filter{
+				{
+					Name:   aws.String("tag:giantswarm.io/cluster"),
+					Values: []*string{aws.String(clusterID)},
+				},
+				{
+					Name:   aws.String("tag:aws:cloudformation:logical-id"),
+					Values: []*string{aws.String("PublicSubnet")},
+				},
+			},
+		}
+
+		o, err := config.AWSClient.EC2.DescribeSubnets(i)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+		if len(o.Subnets) != 1 {
+			return microerror.Maskf(executionFailedError, "expected one subnet, got %d", len(o.Subnets))
+		}
+
+		subnetID = *o.Subnets[0].SubnetId
+		vpcID = *o.Subnets[0].VpcId
+
+		config.Logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("found public subnet %#q and vpc %#q", subnetID, vpcID))
+	}
+
+	var workerSecurityGroupID string
+	{
+		config.Logger.LogCtx(ctx, "level", "debug", "message", "finding worker security group")
+
+		i := &ec2.DescribeSecurityGroupsInput{
+			Filters: []*ec2.Filter{
+				{
+					Name:   aws.String("tag:giantswarm.io/cluster"),
+					Values: []*string{aws.String(clusterID)},
+				},
+				{
+					Name:   aws.String("tag:aws:cloudformation:logical-id"),
+					Values: []*string{aws.String("WorkerSecurityGroup")},
+				},
+			},
+		}
+
+		o, err := config.AWSClient.EC2.DescribeSecurityGroups(i)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+		if len(o.SecurityGroups) != 1 {
+			return microerror.Maskf(executionFailedError, "expected one security group, got %d", len(o.SecurityGroups))
+		}
+
+		workerSecurityGroupID = *o.SecurityGroups[0].GroupId
+
+		config.Logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("found worker security group %#q", workerSecurityGroupID))
+	}
+
+	// We need to create a separate security group in order to allow SSH access to
+	// the bastion instance. The AWS API does not allow tagging the security group
+	// when creating it. That is why we need to separately create tags below, so
+	// we are able to find it later on when we want to clean it up.
+	var bastionSecurityGroupID string
+	{
+		config.Logger.LogCtx(ctx, "level", "debug", "message", "creating bastion security group")
+
+		i := &ec2.CreateSecurityGroupInput{
+			Description: aws.String("Allow SSH access from everywhere to port 22."),
+			GroupName:   aws.String(clusterID + "-bastion"),
```

Comment:
```suggestion
			GroupName:   aws.String("allow-ssh-to-" + clusterID + "-bastion"),
```

Reply:
I do not like the suggestion to change these names. The description is for that. I want the name to be consistent across the resources. There is a pattern and I do not see the added value when prefixing this here.